src/python_routing_v01/compute_nhd_routing_SingleSeg.py

```diff
@@ -302,52 +322,256 @@ def compute_mc_reach_up2down(
             velp=velp,
             depthp=depthp,
         )
-        # print(qdc, velc, depthc)
-        # print(qdc_expected, velc_expected, depthc_expected)
-
-        if write_output:
-            write_buffer.append(
-                ",".join(
-                    map(
-                        str,
-                        (
-                            current_segment,
-                            qdp,
-                            depthp,
-                            velp,
-                            qlat,
-                            qup,
-                            quc,
-                            qdc,
-                            depthc,
-                            velc,
-                        ),
-                    )
-                )
-            )
 
         # for next segment qup / quc use the previous flow values
-        current_flow["flow"]["curr"] = qdc
-        current_flow["depth"]["curr"] = depthc
-        current_flow["vel"]["curr"] = velc
+        if ts > 0:
+            qup = flowveldepth[current_segment]["flowval"][-1]  # input for next segment
+        else:
+            qup = 0
 
-        quc = qdc
-        qup = qdp
+        quc = qdc  # input for next segment
+        if assume_short_ts:
+            quc = qup
 
+        flowveldepth[current_segment]["flowval"].append(qdc)
+        flowveldepth[current_segment]["depthval"].append(depthc)
+        flowveldepth[current_segment]["velval"].append(velc)
+        flowveldepth[current_segment]["time"].append(ts * dt)
+
+        next_segment = connections[current_segment]["downstream"]
         if current_segment == reach["reach_tail"]:
-            if verbose:
+            if debuglevel <= -2:
                 print(f"{current_segment} (tail)")
             break
+        if debuglevel <= -3:
+            print(f"{current_segment} --> {next_segment}\n")
+        # current_segment = next_segment
+        # next_segment = connections[current_segment]["downstream"]
+        current_segment = next_segment
+    # end loop initialized the MC vars
+    # end while loop
+
+
+# ### Psuedocode
+# Write Array  to CSV file
+# arguments reach , pathToOutputFile
+# using global connections and flowveldepth.
+def printarray(
+    reach=None, verbose=False, debuglevel=0, pathToOutputFile="../../test/output/text"
+):
+    global connections
+    global flowveldepth
+
+    # define CSV file Header
+    header = [["time", "qlat", "q", "v", "d"]]
```

Comment:
Why is this a nested list?

Reply:
Because the writerows call expects a nested iterable. If you give it the strings, it iterates them: 
```
"t","i","m","e"
"q","l","a","t"
"q"
"v"
"d"
"0.0","10.0","0.8481932878494263","0.04473819211125374","0.010925279930233955"
```

Reply:
Sorry, I meant to leave a comment below to use `csvwriter.writeheader()`

Reply:
Ok. That makes more sense. Making that change...